docroot/modules/custom/mayflower/js/ajax-pattern-alerts.js

```diff
@@ -166,64 +173,83 @@
+
+        // Don't process if item is not a relevant display.
+        if (
+          item.attributes.field_alert_display !== 'by_organization' &&
+          item.attributes.field_alert_display !== 'specific_target_pages'
+        ) {
+          return;
+        }
+
+        var currentAlertItem = item;
+        var exit = false;
+        var alertReferenceFields;
+        var uuids = [];
+
         if (item.attributes.field_alert_display === 'specific_target_pages') {
-          var currentAlertItem = item;
-          // See if any paragraph in this alert is connected to current page
-          // If yes, we want this alert.
-          currentAlertItem.relationships.field_target_pages_para_ref.data.forEach(function (paraItem) {
-            // NOTE: We have a polyfill to ensure Array.includes() works for us in all browsers.
-            if (paragraphsWithCurrentPageAsTarget.includes(paraItem.id)) {
-              var alertDetailParagraphIds = Drupal.behaviors.mayflower.getAlertParagraphIds(currentAlertItem);
-              var alertDetailParagraphData = Drupal.behaviors.mayflower.getAlertParagraphData(responseData, alertDetailParagraphIds, currentAlertItem);
-              alertDetailParagraphData.forEach(function (alertData) {
-                // NOTE: getAlertParagraphData already sets the id such that it is unique and it changes everytime an alert content is udpated.
-                // We use it to show an alert again, if a user had previously dismissed it, but if the alert now has new updated content.
-                var serializedAlertItem = {
-                  id: alertData.id,
-                  text: alertData.message,
-                  href: alertData.link.href,
-                  info: ''
-                };
-                if (currentAlertItem.attributes.field_alert_severity && currentAlertItem.attributes.field_alert_severity === 'informational_notice') {
-                  serializedAlertItem.prefix = 'Notice';
-                }
-                serializedPageAlertData.headerAlerts.push(serializedAlertItem);
-              });
-              // Exit the loop to prevent multiple copies.
-              return;
-            }
-          });
+          uuids.push(currentPageUuid);
+          alertReferenceFields = currentAlertItem.relationships.field_target_page;
+        }
+        else {
+          // Must be 'by_organization'.
+          uuids = orgPageUuids;
+          alertReferenceFields = currentAlertItem.relationships.field_target_organization;
         }
+
+        // Loop through relevant alert fields.
+        alertReferenceFields.data.forEach(function (paraItem) {
+          if (exit) {
+            return;
+          }
+          // Only get data from the alert if has a relevant uuid to this current page.
+          if ($.inArray(paraItem.id, uuids) !== -1) {
+            var alertDetailParagraphIds = Drupal.behaviors.mayflower.getAlertParagraphIds(currentAlertItem);
```

Comment:
Are we still using paragraphs in other parts of the alerts content type? Or maybe these variable names should be changed. I havent looked too closely.

Reply:
Yes we still need to use the emergency alert paragraph for the alert content. This is a place where using a paragraph is necessary.